SSPSolution/SSPSolution/LevelState.cpp

```diff
@@ -199,6 +199,22 @@ int LevelState::ShutDown()
 	this->m_cHandler->RemoveLastUIComponent();
 	this->m_cHandler->RemoveLastUIComponent();
 
+	/*for (size_t i = 0; i < this->m_Player1ChainPhysicsComp.size(); i++)
+	{
+		delete this->m_Player1ChainPhysicsComp[i];
+		this->m_Player1ChainPhysicsComp[i] = nullptr;
+	}
+	for (size_t i = 0; i < this->m_Player2ChainPhysicsComp.size(); i++)
+	{
+		delete this->m_Player2ChainPhysicsComp[i];
+		this->m_Player2ChainPhysicsComp[i] = nullptr;
+	}*/
```

Comment:
If it is truly handled in the code below that clears the vectors then I suggest that you remove this out commented code to avoid confusion.